src/loaders/agent.js

```diff
@@ -0,0 +1,56 @@
+// loader files
+import { getEnabledFeatures } from './features/enabled-features'
+import { configure } from './configure/configure'
+import { getFeatureDependencyNames } from './features/featureDependencies'
+import { featurePriority } from './features/features'
+// common files
+import { Aggregator } from '../common/aggregate/aggregator'
+import { gosNREUMInitializedAgents } from '../common/window/nreum'
+import { generateRandomHexString } from '../common/ids/unique-id'
+import { getConfiguration, getInfo, getLoaderConfig, getRuntime } from '../common/config/config'
+
+
+export class Agent {
+    constructor(options, agentIdentifier = generateRandomHexString(16)) {
+        this.agentIdentifier = agentIdentifier
+        this.sharedAggregator = new Aggregator({ agentIdentifier: this.agentIdentifier })
+        this.features = {}
+
+        this.desiredFeatures = options.features || []
+        this.desiredFeatures.sort((a, b) => featurePriority[a.featureName] - featurePriority[b.featureName])
+
+        Object.assign(this, configure(this.agentIdentifier, options, options.loaderType || 'agent'))
+
+        this.start()
+    }
+
+    get config() {
+        return {
+            info: getInfo(this.agentIdentifier),
+            init: getConfiguration(this.agentIdentifier),
+            loader_config: getLoaderConfig(this.agentIdentifier),
+            runtime: getRuntime(this.agentIdentifier)
+        }
+    }
+
+    start() {
+        try {
+            const enabledFeatures = getEnabledFeatures(this.agentIdentifier)
+            this.desiredFeatures.forEach(f => {
+                if (enabledFeatures[f.featureName]) {
+                    const dependencies = getFeatureDependencyNames(f.featureName)
```

Comment:
Just a 👍 on this. It's a cleaner iterative-style rework of the former system. May break if the `featurePriority` order that's tied to the dependency tree changes, but for now, this is satisfactory.